libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/postgre/PostgresSchemaAccessor.java

```diff
@@ -0,0 +1,388 @@
+import com.oceanbase.tools.dbbrowser.model.DBPackage;
+import com.oceanbase.tools.dbbrowser.model.DBProcedure;
+import com.oceanbase.tools.dbbrowser.model.DBSequence;
+import com.oceanbase.tools.dbbrowser.model.DBSynonym;
+import com.oceanbase.tools.dbbrowser.model.DBSynonymType;
+import com.oceanbase.tools.dbbrowser.model.DBTable;
+import com.oceanbase.tools.dbbrowser.model.DBTable.DBTableOptions;
+import com.oceanbase.tools.dbbrowser.model.DBTableColumn;
+import com.oceanbase.tools.dbbrowser.model.DBTableConstraint;
+import com.oceanbase.tools.dbbrowser.model.DBTableIndex;
+import com.oceanbase.tools.dbbrowser.model.DBTablePartition;
+import com.oceanbase.tools.dbbrowser.model.DBTableSubpartitionDefinition;
+import com.oceanbase.tools.dbbrowser.model.DBTrigger;
+import com.oceanbase.tools.dbbrowser.model.DBType;
+import com.oceanbase.tools.dbbrowser.model.DBVariable;
+import com.oceanbase.tools.dbbrowser.model.DBView;
+import com.oceanbase.tools.dbbrowser.schema.DBSchemaAccessor;
+import com.oceanbase.tools.dbbrowser.util.StringUtils;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class PostgresSchemaAccessor implements DBSchemaAccessor {
+
+    protected JdbcOperations jdbcOperations;
+
+    public PostgresSchemaAccessor(@NonNull JdbcOperations jdbcOperations) {
+        this.jdbcOperations = jdbcOperations;
+    }
+
+    @Override
+    public List<String> showDatabases() {
+        String sql = "SELECT datname FROM pg_database;";
+        return jdbcOperations.queryForList(sql, String.class);
+    }
+
+    @Override
+    public DBDatabase getDatabase(String schemaName) {
+        throw new UnsupportedOperationException("Not supported yet");
+    }
+
+    @Override
+    public List<DBDatabase> listDatabases() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("SELECT datname AS database_name,pg_encoding_to_char(encoding) AS character_set,"
+                + "datcollate AS collation FROM pg_database;");
+        return jdbcOperations.query(sb.toString(), (rs, rowNum) -> {
+            DBDatabase database = new DBDatabase();
+            database.setId(rs.getString("database_name"));
+            database.setName(rs.getString("database_name"));
+            database.setCharset(rs.getString("character_set"));
+            database.setCollation(rs.getString("collation"));
+            return database;
+        });
+    }
+
+    @Override
+    public void switchDatabase(String schemaName) {
+        throw new UnsupportedOperationException("Not supported yet");
+    }
+
+    @Override
+    public List<DBObjectIdentity> listUsers() {
+        throw new UnsupportedOperationException("Not supported yet");
+    }
+
+    @Override
+    public List<String> showTables(String schemaName) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("SELECT table_name FROM information_schema.tables "
+                + "WHERE table_schema = 'public' AND table_type = 'BASE TABLE'");
```

Comment:
why should we use 'public' as the schema?

Reply:
In the Postgres database, the public schema serves as the repository for user-defined data objects. In contrast, other schemas such as pg_catalog schema contain metadata related to the database system, and the information_schema holds information regarding the entire database. Here, as we anticipate presenting user-defined data objects, we employ the public schema.